### ISSUE TITLE
CDDLib Breaks NV.jl

### ISSUE THREAD

**Issue by chelseas** [sisl/NeuralVerification.jl#41]
Julia version: 1.0.2
I tried to use NV.jl:
```
using NeuralVerification
```

And then I got this error:
```
ERROR: LoadError: CDDLib not properly installed. Please run Pkg.build("CDDLib")
```

So I ran 
```
(v1.0) pkg> build CDDLib 
```
And I got this:
```
┌ Error: Error building `CDDLib`, showing the last 100 of log: 
<skipping a bunch of stuff>
│ Warning: The compilation of cddlib requires the header gmp.h provided by the package libgmp-dev.
│ If the compilation fails, please install it as follows:
│ $ sudo apt-get install libgmp-dev
 ```

So I ran 

```sudo apt-get install libgmp-dev```

 successfully. 

But then I still got the same errors. Same "no gmp.h" stuff.

**Comment by tomerarnon** [comment]
Haven't seen that issue before. Have you tried removing and reinstalling CDDLib (the turn-it-off-and-on-again of julia packages...)

**Comment by kjulian3** [comment]
Is there a solution to this issue? I am having the same problem. I installed libgmp-dev, but I'm still getting the error when I try to build CDDLib.

**Comment by tomerarnon** [comment]
As far as I know, no. I've never seen this issue myself, and haven't investigated it. @chelseas may have found a way to resolve it?

**Comment by kjulian3** [comment]
I had to add the libgmp.so file to my LD_PRELOAD, as described here: https://github.com/JuliaPolyhedra/CDDLib.jl/issues/26

Now it's working for me.

**Comment by ksilken** [comment]
I'm trying to use NeuralVerification but I've run into a few problems

`julia> using NeuralVerification
[ Info: Precompiling NeuralVerification [146f25fa-00e7-11e9-3ae5-fdbac6e12fa7]
ERROR: LoadError: CDDLib not properly installed. Please run Pkg.build("CDDLib")
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] top-level scope at /Users/KevinSilken/.julia/packages/CDDLib/bKo5p/src/CDDLib.jl:12
 [3] include at ./boot.jl:328 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1105
 [5] include(::Module, ::String) at ./Base.jl:31
 [6] top-level scope at none:2
 [7] eval at ./boot.jl:330 [inlined]
 [8] eval(::Expr) at ./client.jl:425
 [9] top-level scope at ./none:3
in expression starting at /Users/KevinSilken/.julia/packages/CDDLib/bKo5p/src/CDDLib.jl:9
ERROR: LoadError: Failed to precompile CDDLib [3391f64e-dcde-5f30-b752-e11513730f60] to /Users/KevinSilken/.julia/compiled/v1.3/CDDLib/SMSjF_9qgMz.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1283
 [3] _require(::Base.PkgId) at ./loading.jl:1024
 [4] require(::Base.PkgId) at ./loading.jl:922
 [5] require(::Module, ::Symbol) at ./loading.jl:917
 [6] include at ./boot.jl:328 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1105
 [8] include(::Module, ::String) at ./Base.jl:31
 [9] top-level scope at none:2
 [10] eval at ./boot.jl:330 [inlined]
 [11] eval(::Expr) at ./client.jl:425
 [12] top-level scope at ./none:3
in expression starting at /Users/KevinSilken/.julia/packages/NeuralVerification/4qLbJ/src/NeuralVerification.jl:8
ERROR: Failed to precompile NeuralVerification [146f25fa-00e7-11e9-3ae5-fdbac6e12fa7] to /Users/KevinSilken/.julia/compiled/v1.3/NeuralVerification/sJvOT_9qgMz.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1283
 [3] _require(::Base.PkgId) at ./loading.jl:1024
 [4] require(::Base.PkgId) at ./loading.jl:922
 [5] require(::Module, ::Symbol) at ./loading.jl:917

julia> 
`

So I did:

`(v1.3) pkg> build CDDLib
  Building LibCURL ─→ `~/.julia/packages/LibCURL/lWJxD/deps/build.log`
  Building WinRPM ──→ `~/.julia/packages/WinRPM/Y9QdZ/deps/build.log`
  Building Homebrew → `~/.julia/packages/Homebrew/s09IX/deps/build.log`
┌ Error: Error building `Homebrew`: 
│ /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- plist (LoadError)
│ 	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/system_command.rb:3:in `<top (required)>'
│ 	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
│ 	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/global.rb:14:in `<top (required)>'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/brew.rb:15:in `require_relative'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/brew.rb:15:in `<main>'
│ ERROR: LoadError: failed process: Process(`/Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/bin/brew update --force`, ProcessExited(1)) [1]
│ 
│ Stacktrace:
│  [1] pipeline_error at ./process.jl:525 [inlined]
│  [2] #run#565(::Bool, ::typeof(run), ::Cmd) at ./process.jl:440
│  [3] run at ./process.jl:438 [inlined]
│  [4] #brew#4(::Bool, ::Bool, ::Bool, ::Bool, ::Bool, ::typeof(Homebrew.brew), ::Cmd) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:19
│  [5] #brew at ./tuple.jl:0 [inlined]
│  [6] #update#6(::Bool, ::typeof(Homebrew.update)) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:50
│  [7] update() at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:50
│  [8] top-level scope at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/build.jl:3
│  [9] include at ./boot.jl:328 [inlined]
│  [10] include_relative(::Module, ::String) at ./loading.jl:1105
│  [11] include(::Module, ::String) at ./Base.jl:31
│  [12] include(::String) at ./client.jl:424
│  [13] top-level scope at none:5
│ in expression starting at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/build.jl:1
└ @ Pkg.Operations /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/backwards_compatible_isolation.jl:649
  Building CDDLib ──→ `~/.julia/packages/CDDLib/bKo5p/deps/build.log`
┌ Error: Error building `CDDLib`: 
│ /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- plist (LoadError)
│ 	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/system_command.rb:3:in `<top (required)>'
│ 	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
│ 	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/global.rb:14:in `<top (required)>'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/brew.rb:15:in `require_relative'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/brew.rb:15:in `<main>'
│ ERROR: LoadError: ArgumentError: `brew info` failed for ["gmp"]!
│ Stacktrace:
│  [1] json(::Array{String,1}) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:192
│  [2] json at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:214 [inlined]
│  [3] #direct_deps#10(::Bool, ::typeof(Homebrew.direct_deps), ::String) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:276
│  [4] #direct_deps at ./none:0 [inlined]
│  [5] #deps_tree#17(::Bool, ::typeof(Homebrew.deps_tree), ::String) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:306
│  [6] #deps_tree at ./tuple.jl:0 [inlined]
│  [7] #deps_sorted#24(::Bool, ::typeof(Homebrew.deps_sorted), ::String) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:379
│  [8] deps_sorted at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:379 [inlined]
│  [9] #add#29(::Bool, ::Bool, ::typeof(Homebrew.add), ::String) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:420
│  [10] add(::String) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:412
│  [11] top-level scope at /Users/KevinSilken/.julia/packages/CDDLib/bKo5p/deps/build.jl:38
│  [12] include at ./boot.jl:328 [inlined]
│  [13] include_relative(::Module, ::String) at ./loading.jl:1105
│  [14] include(::Module, ::String) at ./Base.jl:31
│  [15] include(::String) at ./client.jl:424
│  [16] top-level scope at none:5
│ in expression starting at /Users/KevinSilken/.julia/packages/CDDLib/bKo5p/deps/build.jl:36
│ caused by [exception 1]
│ failed process: Process(`/Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/bin/brew info --json=v1 gmp`, ProcessExited(1)) [1]
│ 
│ Stacktrace:
│  [1] pipeline_error at ./process.jl:525 [inlined]
│  [2] read(::Cmd) at ./process.jl:412
│  [3] read(::Cmd, ::Type{String}) at ./process.jl:421
│  [4] readchomp(::Cmd) at ./io.jl:873
│  [5] #brewchomp#5(::Bool, ::Bool, ::Bool, ::Bool, ::Bool, ::typeof(Homebrew.brewchomp), ::Cmd) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:39
│  [6] brewchomp at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:31 [inlined]
│  [7] json(::Array{String,1}) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:186
│  [8] json at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:214 [inlined]
│  [9] #direct_deps#10(::Bool, ::typeof(Homebrew.direct_deps), ::String) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:276
│  [10] #direct_deps at ./none:0 [inlined]
│  [11] #deps_tree#17(::Bool, ::typeof(Homebrew.deps_tree), ::String) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:306
│  [12] #deps_tree at ./tuple.jl:0 [inlined]
│  [13] #deps_sorted#24(::Bool, ::typeof(Homebrew.deps_sorted), ::String) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:379
│  [14] deps_sorted at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:379 [inlined]
│  [15] #add#29(::Bool, ::Bool, ::typeof(Homebrew.add), ::String) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:420
│  [16] add(::String) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:412
│  [17] top-level scope at /Users/KevinSilken/.julia/packages/CDDLib/bKo5p/deps/build.jl:38
│  [18] include at ./boot.jl:328 [inlined]
│  [19] include_relative(::Module, ::String) at ./loading.jl:1105
│  [20] include(::Module, ::String) at ./Base.jl:31
│  [21] include(::String) at ./client.jl:424
│  [22] top-level scope at none:5
└ @ Pkg.Operations /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/backwards_compatible_isolation.jl:649
`
Then I tried updating, adding homebrew, updating again, and building homebrew, but:
`(v1.3) pkg> build Homebrew
  Building Homebrew → `~/.julia/packages/Homebrew/s09IX/deps/build.log`
┌ Error: Error building `Homebrew`: 
│ /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- plist (LoadError)
│ 	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/system_command.rb:3:in `<top (required)>'
│ 	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
│ 	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/global.rb:14:in `<top (required)>'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/brew.rb:15:in `require_relative'
│ 	from /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/Library/Homebrew/brew.rb:15:in `<main>'
│ ERROR: LoadError: failed process: Process(`/Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/usr/bin/brew update --force`, ProcessExited(1)) [1]
│ 
│ Stacktrace:
│  [1] pipeline_error at ./process.jl:525 [inlined]
│  [2] #run#565(::Bool, ::typeof(run), ::Cmd) at ./process.jl:440
│  [3] run at ./process.jl:438 [inlined]
│  [4] #brew#4(::Bool, ::Bool, ::Bool, ::Bool, ::Bool, ::typeof(Homebrew.brew), ::Cmd) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:19
│  [5] #brew at ./tuple.jl:0 [inlined]
│  [6] #update#6(::Bool, ::typeof(Homebrew.update)) at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:50
│  [7] update() at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/src/API.jl:50
│  [8] top-level scope at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/build.jl:3
│  [9] include at ./boot.jl:328 [inlined]
│  [10] include_relative(::Module, ::String) at ./loading.jl:1105
│  [11] include(::Module, ::String) at ./Base.jl:31
│  [12] include(::String) at ./client.jl:424
│  [13] top-level scope at none:5
│ in expression starting at /Users/KevinSilken/.julia/packages/Homebrew/s09IX/deps/build.jl:1
└ @ Pkg.Operations /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/backwards_compatible_isolation.jl:649

(v1.3) pkg> 
`
Any suggestions?

**Comment by tomerarnon** [comment]
Unfortunately, I don't have anything up my sleeve. I just tried adding and building `CDDLib` in a fresh julia 1.3 and was able to do it successfully. Then, I tried doing the same with `Homebrew` and failed (differently from you though). But I don't know why buildling `CDDLib` worked for me and not for you, nor why either of our homebrews are failing.

I'm not sure which of those packages would be more promising to reach out to for help (maybe hedge your bets and go with both). My CDDLib build might have been influenced by the existence of a previously compiled cddlib from my julia 1.0/1.2 versions, and that's why it wasn't going through homebrew like yours? No idea...
